### PR TITLE
WIP: strategy: pivotshort: pivot low shorting strategy

### PR DESCRIPTION
**Pull request by austin362667**
```
1h * 100 for finding Pivot Low

SL: 0.8%,  Avg SL: twice per profit trade
TP: 12% (15*SL)

BounceRatio: 3% for posting LO
```

### CODE REVIEW

**Comment by austin362667** [comment]
telegram-cloud-photo-size-5-6325772007467234253-y

File: pkg/strategy/pivot/strategy.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	Position       *types.Position  `json:"position,omitempty"`
+	StopLossRatio  fixedpoint.Value `json:"stopLossRatio"`
+	CatBounceRatio fixedpoint.Value `json:"catBounceRatio"`
+
+	activeMakerOrders *bbgo.LocalActiveOrderBook
+	orderStore        *bbgo.OrderStore
+	tradeCollector    *bbgo.TradeCollector
+
+	session *bbgo.ExchangeSession
+	book    *types.StreamOrderBook
```
Comment (c9s) [inline pkg/strategy/pivot/strategy.go:41]: you don't need this field?

File: pkg/strategy/pivot/strategy.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+		}
+		log.Info(s.pivot.LastLow())
+		if s.pivot.LastLow() > 0. {
+
+			lastLow = fixedpoint.NewFromFloat(s.pivot.LastLow())
+
+		} else {
+			lastLow = fixedpoint.Zero
+			// SL || TP
+			if kline.Close.Div(s.Position.AverageCost).Compare(fixedpoint.One.Add(s.StopLossRatio)) > 0 || kline.Close.Div(s.Position.AverageCost).Compare(fixedpoint.One.Sub(s.StopLossRatio.Mul(fixedpoint.NewFromInt(15)))) < 0 {
```
Comment (c9s) [inline pkg/strategy/pivot/strategy.go:142]: pull `kline.Close.Div(s.Position.AverageCost)` as a variable?

File: pkg/cmd/builtin.go [hunk 3]
```diff
@@ -13,6 +13,7 @@ import (
 	_ "github.com/c9s/bbgo/pkg/strategy/funding"
 	_ "github.com/c9s/bbgo/pkg/strategy/grid"
 	_ "github.com/c9s/bbgo/pkg/strategy/kline"
+	_ "github.com/c9s/bbgo/pkg/strategy/pivot"
```
Comment (c9s) [inline pkg/cmd/builtin.go:16]: name this strategy `pivotshort` ?

Comment (austin362667) [reply pkg/cmd/builtin.go:16]: sure!

File: pkg/strategy/pivotshort/pivot.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
+package pivotshort
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/indicator"
```
Comment (zenixls2) [inline pkg/strategy/pivotshort/pivot.go:7]: maybe move this file to indicator?

Comment (zenixls2) [reply pkg/strategy/pivotshort/pivot.go:7]: also add at least one test case for pivot

**Review by zenixls2 — APPROVED** [review]

**Review by c9s — APPROVED** [review]

**Pull request merged**